rusk/src/lib/node/rusk.rs

```diff
@@ -206,16 +204,7 @@ impl Rusk {
                     let err = receipt.data.err().map(|e| format!("{e}"));
                     info!("Tx {tx_id_hex} executed with {gas_spent} gas and err {err:?}");
 
-                    let tx_events: Vec<_> = receipt
-                        .events
-                        .into_iter()
-                        .map(|event| ContractTxEvent {
-                            event: event.into(),
-                            origin: Some(tx_id),
```

Comment:
Any reason to exclude the tx_id that generated the event?

Reply:
Why was it included in the first place? 🤔 Does the client need assurances that the event happens due to a particular transaction? If so, under what circumstances. Given some, https://github.com/dusk-network/rusk/pull/2539#discussion_r1781410884, already seem to think that including the whole data is too much, I hesitate to put more into the bloom.

Reply:
We could place it all into the bloom:

- `(contract, topic)`
- `(contract, topic, origin)`
- `(contract, topic, origin, data)`

But at that point why not all the possible permutations? It starts to feel like overkill.

Reply:
@Neotamandua I understand you use the origin field for archivals. How useful is it to have it in the bloom?

Reply:
I believe we must insert the wider one, and if I need to search for a specific subset I've to ask to some archival about all the missing info, and once retrieved the data then check the bloom filter

Does that make sense?

Reply:
I had a talk with @Neotamandua and we were fully convinced that only the `(contract, topic)` use can can *only* be provided by the filter. Anything else can be provided by other means, be it archivals or pure re-execution of the block. We will merge it like that, and deal with any change we might want to make during the testnet window.